--- /usr/share/perl5/Munin/Master/Config.pm.orig	2019-05-23 19:14:44.795386000 -0400
+++ /usr/share/perl5/Munin/Master/Config.pm	2019-05-23 19:15:00.397955000 -0400
@@ -456,7 +456,7 @@
 	    " Unknown keyword at end of left hand side of line $NR ($key $value)\n";
     }
 
-    if ($host =~ /[^-A-Za-z0-9\.]/) {
+    if ($host =~ /[^-_A-Za-z0-9\.]/) {
 	# Since we're not quite sure what context we're called in we'll report the error message more times rather than fewer.
 	ERROR "[ERROR] Hostname '$host' contains illegal characters (http://en.wikipedia.org/wiki/Hostname#Restrictions_on_valid_host_names).  Please fix this by replacing illegal characters with '-'.  Remember to do it on both in the master configuration and on the munin-node.";
 	croak "[ERROR] Hostname '$host' contains illegal characters (http://en.wikipedia.org/wiki/Hostname#Restrictions_on_valid_host_names).  Please fix this by replacing illegal characters with '-'.  Remember to do it on both in the master configuration and on the munin-node.\n";
