--- /usr/share/perl5/Munin/Node/Server.pm.orig	2019-05-23 18:58:13.677053000 -0400
+++ /usr/share/perl5/Munin/Node/Server.pm	2019-05-23 18:53:34.665350000 -0400
@@ -167,7 +167,7 @@
     }
 
     logger ("DEBUG: Running command '$_'.") if $config->{DEBUG};
-    if (/^list\s*([0-9a-zA-Z\.\-]+)?/i) {
+    if (/^list\s*([0-9a-zA-Z\.\-_]+)?/i) {
 	my $hostname_lc = defined($1) ? lc($1) : undef;
         _list_services($session, $hostname_lc);
     }
